test/icinga-notification.cpp

```diff
@@ -102,4 +149,63 @@ BOOST_AUTO_TEST_CASE(type_filter)
 	std::cout << "#4 Notification type: " << ftype << " against " << notification->GetTypeFilter() << " must fail." << std::endl;
 	BOOST_CHECK(!(notification->GetTypeFilter() & ftype));
 }
+
+BOOST_AUTO_TEST_CASE(duplicate_dueto_filter)
+{
+	DuplicateDueToFilterHelper helper;
+
+	helper.s->SetState(ServiceCritical, true);
+	Application::GetTP().Start();
+	helper.n->BeginExecuteNotification(NotificationProblem, nullptr, false, false, "", "");
+	Application::GetTP().Stop();
+	BOOST_CHECK(helper.called == 1u);
+
+	helper.s->SetState(ServiceWarning, true);
+	Application::GetTP().Start();
+	helper.n->BeginExecuteNotification(NotificationProblem, nullptr, false, false, "", "");
+	Application::GetTP().Stop();
+	BOOST_CHECK(helper.called == 1u);
+
+	helper.s->SetState(ServiceCritical, true);
+	Application::GetTP().Start();
+	helper.n->BeginExecuteNotification(NotificationProblem, nullptr, false, false, "", "");
+	Application::GetTP().Stop();
+	BOOST_CHECK(helper.called == 1u);
+}
+
+BOOST_AUTO_TEST_CASE(duplicate_dueto_filter_recovery)
+{
+	DuplicateDueToFilterHelper helper;
+
+	helper.s->SetState(ServiceCritical, true);
+	Application::GetTP().Start();
+	helper.n->BeginExecuteNotification(NotificationProblem, nullptr, false, false, "", "");
+	Application::GetTP().Stop();
+	BOOST_CHECK(helper.called == 1u);
+
+	helper.s->SetState(ServiceWarning, true);
+	Application::GetTP().Start();
+	helper.n->BeginExecuteNotification(NotificationProblem, nullptr, false, false, "", "");
+	Application::GetTP().Stop();
+	BOOST_CHECK(helper.called == 1u);
+
+	helper.s->SetState(ServiceCritical, true);
+	Application::GetTP().Start();
+	helper.n->BeginExecuteNotification(NotificationProblem, nullptr, false, false, "", "");
+	Application::GetTP().Stop();
+	BOOST_CHECK(helper.called == 1u);
+
+	helper.s->SetState(ServiceOK, true);
+	Application::GetTP().Start();
+	helper.n->BeginExecuteNotification(NotificationRecovery, nullptr, false, false, "", "");
+	Application::GetTP().Stop();
+	BOOST_CHECK(helper.called == 1u);
```

Comment:
It might also be worth adding this test here:
```c++
BOOST_CHECK(helper.n->GetLastNotifiedStatePerUser()->Get(helper.u->GetName()) == Empty);
```

Reply:
Redundant IMAO. It's only the bridge to (not) sending a notification, the latter is important and tested here.